kernel/src/fs/path/mount.rs

```diff
@@ -8,28 +10,28 @@ use crate::{
     prelude::*,
 };
 
-/// The MountNode can form a mount tree to maintain the mount information.
+/// The `MountNode` is used to form a mount tree to maintain the mount information.
 pub struct MountNode {
-    /// Root Dentry_.
+    /// Root dentry.
     root_dentry: Arc<Dentry_>,
-    /// Mountpoint Dentry_. A mount node can be mounted on one dentry of another mount node,
+    /// Mountpoint dentry. A mount node can be mounted on one dentry of another mount node,
     /// which makes the mount being the child of the mount node.
-    mountpoint_dentry: RwLock<Option<Arc<Dentry_>>>,
+    mountpoint_dentry: RwMutex<Option<Arc<Dentry_>>>,
     /// The associated FS.
     fs: Arc<dyn FileSystem>,
     /// The parent mount node.
-    parent: RwLock<Option<Weak<MountNode>>>,
+    parent: RwMutex<Option<Weak<MountNode>>>,
     /// Child mount nodes which are mounted on one dentry of self.
-    children: Mutex<BTreeMap<DentryKey, Arc<Self>>>,
+    children: RwMutex<HashMap<DentryKey, Arc<Self>>>,
```

Comment:
Read-write locks should only be used when we are certain that writes lock are seldomly acquired. But creating files could be quite frequent for some workloads. So in the case of `children`, I guess using `Mutex` is more appropriate than `RwMutex`. Do you have any other considerations?

Reply:
I actually think through this problem. I think this `children` in `MountNode` can be considered read-most. Acquiring its write lock only appears in `mount()`/`umount()` (not file creation/deletion), which is much less rare than file manipulates. While acquiring its read lock appears in every lookup in the `Dentry` of the root `MountNode`.